src/internal/http/listener/httpslistener_test.go

```diff
@@ -0,0 +1,229 @@
+package httpslistener_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/dfds/confluent-gateway/configuration"
+	httpslistener "github.com/dfds/confluent-gateway/internal/http/listener"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	topic1 = "i.exist.and.people.listen.to.me"
+	topic2 = "i.also.exist.and.a.thing.listens"
+
+	consumer1 = "consumer1-prod"
+	consumer2 = "consumer2-prod"
+	consumer3 = "consumer3-prod"
+	consumer4 = "consumer4-dev"
+)
+
+type channelBuffer struct {
+	cin  chan bool
+	cout chan bool
+}
+
+func (cb *channelBuffer) Run() {
+	go func() {
+		<-cb.cin
+		cb.cout <- true
+	}()
+}
+
+type mockConsumerGroupHandler struct{}
+
+func (m *mockConsumerGroupHandler) Setup(session sarama.ConsumerGroupSession) error { return nil }
+func (m *mockConsumerGroupHandler) Cleanup(sarama.ConsumerGroupSession) error       { return nil }
+func (m *mockConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
+	for message := range claim.Messages() {
+		session.MarkMessage(message, "")
+		//session.Commit()
+	}
+	return nil
+}
```

Comment:
It would be helpful with a comment here that explains the flow of the session if you, like me, is looking at this code without too much knowledge of sarama/kafka